src/TimeTracker.Library/Services/TimeEntryService.cs

```diff
@@ -63,11 +62,22 @@ public async Task<Guid> CreateBillableTimeEntry(DateTime date, double hours, int
 
         public async Task<double> DeleteHours(DateTime date)
         {
-            var timeEntries = await db.TimeEntries.Where(x => x.UserId == userId && x.Date >= date.Date).ToListAsync();
+            var cutOffDate = DateTime.UtcNow.Date.AddHours(-48);
+            if (date < cutOffDate)
+            {
+                throw new Exception("Entries older than 48 hours cannot be deleted.");
+            }
+            
+            var timeEntries = await db.TimeEntries.Where(x => x.UserId == userId && 
+                                                              x.Date.Date == date.Date.Date && 
+                                                              x.Date >= cutOffDate).ToListAsync();
```

Comment:
Do you want to floor this to just the .Date part as well?

Reply:
... `.Date` floors the time and leaves the day